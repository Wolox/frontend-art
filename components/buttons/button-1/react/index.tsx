import React from 'react';

import '../styles.scss';

function Button1() {
  return (
    <div className="column center btn-group-1">
      <button className="btn btn-1 m-bottom-2" disabled={true}>
        <div className="btn__content">cancel</div>
      </button>
      <button type="button" className="btn btn-1">
        <div className="btn__content">submit</div>
      </button>
    </div>
  );
}

export default Button1;
