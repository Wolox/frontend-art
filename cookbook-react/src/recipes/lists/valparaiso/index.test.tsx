import React from 'react';
import { render, waitFor, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Example from '.';

beforeEach(() => {
  jest.useFakeTimers();
});

test('Should match snapshot', () => {
  const { container } = render(<Example />);
  expect(container).toMatchSnapshot();
});

test('Should match snapshot', async () => {
  render(<Example />);

  const buttons = screen.getAllByRole('button');
  const firstButton = buttons[0];

  userEvent.click(firstButton);
  await waitFor(() => jest.runAllTimers());

  expect(firstButton.style.height).toBeFalsy();
});

afterEach(() => {
  jest.useRealTimers();
});
