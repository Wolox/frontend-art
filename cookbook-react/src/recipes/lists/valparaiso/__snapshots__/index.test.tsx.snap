// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<div>
  <div
    class="exampleContainer"
  >
    <div
      class="itemContainer m-bottom-3"
    >
      <label
        class="m-bottom-2 title"
      >
        Listado de spans
      </label>
      <ul
        class="column"
      >
        <span
          class="m-bottom-1"
        >
          Item 1
        </span>
        <span
          class="m-bottom-1"
        >
          Item 2
        </span>
        <span
          class="m-bottom-1"
        >
          Item 3
        </span>
        <span
          class="m-bottom-1"
        >
          Item 4
        </span>
        <span
          class="m-bottom-1"
        >
          Item 5
        </span>
        <span
          class="m-bottom-1"
        >
          Item 6
        </span>
      </ul>
    </div>
    <div
      class="itemContainer m-bottom-3"
    >
      <label
        class="m-bottom-2 title"
      >
        Listado de custom items
      </label>
      <ul
        class="column"
      >
        <li
          class="listItem row middle space-between"
          style="transition: height 300ms ease-out, padding 300ms ease-out, margin 300ms ease-out; overflow: hidden; height: 0px;"
        >
          <div
            class="item-1"
          >
            Item 1
          </div>
          <button
            class="closeIcon"
            type="button"
          />
        </li>
        <li
          class="listItem row middle space-between"
          style="transition: height 300ms ease-out, padding 300ms ease-out, margin 300ms ease-out; overflow: hidden; height: 0px;"
        >
          <div
            class="item-1"
          >
            Item 2
          </div>
          <button
            class="closeIcon"
            type="button"
          />
        </li>
        <li
          class="listItem row middle space-between"
          style="transition: height 300ms ease-out, padding 300ms ease-out, margin 300ms ease-out; overflow: hidden; height: 0px;"
        >
          <div
            class="item-1"
          >
            Item 3
          </div>
          <button
            class="closeIcon"
            type="button"
          />
        </li>
        <li
          class="listItem row middle space-between"
          style="transition: height 300ms ease-out, padding 300ms ease-out, margin 300ms ease-out; overflow: hidden; height: 0px;"
        >
          <div
            class="item-1"
          >
            Item 4
          </div>
          <button
            class="closeIcon"
            type="button"
          />
        </li>
        <li
          class="listItem row middle space-between"
          style="transition: height 300ms ease-out, padding 300ms ease-out, margin 300ms ease-out; overflow: hidden; height: 0px;"
        >
          <div
            class="item-1"
          >
            Item 5
          </div>
          <button
            class="closeIcon"
            type="button"
          />
        </li>
        <li
          class="listItem row middle space-between"
          style="transition: height 300ms ease-out, padding 300ms ease-out, margin 300ms ease-out; overflow: hidden; height: 0px;"
        >
          <div
            class="item-1"
          >
            Item 6
          </div>
          <button
            class="closeIcon"
            type="button"
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;
